client/src/partials/purchases/templates/cellStatus.tmpl.html

```diff
@@ -0,0 +1,17 @@
+<div class="ui-grid-cell-contents">
+    <div ng-if="row.entity.is_confirmed === 1 && row.entity.is_cancelled === 0 && row.entity.is_received === 0" class="label label-primary" translate>
```

Comment:
It seems to me that you could reduce the number of checks here by nesting them so that angular doesn't execute all these checks on every change.